sdk/java/dagger-java-annotation-processor/src/main/java/io/dagger/annotation/processor/DaggerModuleAnnotationProcessor.java

```diff
@@ -513,11 +610,15 @@ private static CodeBlock functionInvoke(ObjectInfo objectInfo, FunctionInfo fnIn
 
   public static CodeBlock withFunction(ObjectInfo objectInfo, FunctionInfo fnInfo)
       throws ClassNotFoundException {
+    boolean isConstructor = fnInfo.name().equals("<init>");
     CodeBlock.Builder code =
         CodeBlock.builder()
-            .add("\n                dag.function($S,", fnInfo.name())
+            .add("\n                dag.function($S,", isConstructor ? "New" : fnInfo.name())
```

Comment:
```suggestion
            .add("\n                dag.function($S,", isConstructor ? "" : fnInfo.name())
```
See:

https://github.com/dagger/dagger/blob/469639e6be403eea3e1a2cc7044a99a2b2523005/sdk/python/src/dagger/mod/_module.py#L170-L171

https://github.com/dagger/dagger/blob/469639e6be403eea3e1a2cc7044a99a2b2523005/sdk/python/src/dagger/mod/_resolver.py#L219-L220